Objects/genericaliasobject.c

```diff
@@ -487,11 +487,49 @@ ga_reduce(PyObject *self, PyObject *Py_UNUSED(ignored))
                          alias->origin, alias->args);
 }
 
+static PyObject *
+ga_dir(PyObject *self, PyObject *Py_UNUSED(ignored))
+{
+    gaobject *alias = (gaobject *)self;
+    PyObject *dir = PyObject_Dir(alias->origin);
+    if (dir == NULL) {
+        return NULL;
+    }
+
+    int contains = 0;
+    PyObject *dir_entry = NULL;
+    for (const char * const *p = attr_exceptions; ; p++) {
+        if (*p == NULL) {
+            break;
+        }
+        else {
+            PyObject *dir_entry = PyUnicode_FromString(*p);
+            if (dir_entry == NULL) {
+                goto error;
+            } else if ((contains = PySequence_Contains(dir, dir_entry)) < 0) {
+                goto error;
+            }
+
+            if (contains == 0 && PyList_Append(dir, dir_entry) < 0) {
+                goto error;
+            }
+            Py_DECREF(dir_entry);
```

Comment:
Just to be safe I'd use `Py_CLEAR`.

(Another way out of this would be to get rid of the error-cleanup for this variable and instead decref it in the two `goto error` blocks above.)